vivarium/core/serialize.py

```diff
@@ -184,113 +82,28 @@ def deserialize(self, data: dict) -> dict:
         }
 
 
-class NumpyScalarSerializer(Serializer):
-    """Serializer for Numpy scalars.
-
-    Note that this serialization is lossy. Upon deserialization, the
-    serialized values will remain Python builtin types, not Numpy
-    scalars.
-    """
-
-    def __init__(self) -> None:
-        super().__init__(exclusive_types=(
-            np.bool_,
-            np.int8, np.int16, np.int32, np.int64,
-            np.uint8, np.uint16, np.uint32, np.uint64,
-            np.float16, np.float32, np.float64,
-            np.complex64, np.complex128,
-        ))
-
-    @staticmethod
-    def can_serialize(data: Any) -> bool:
-        return isinstance(data, (np.integer, np.floating, np.bool_))
-
-    def serialize(
-            self,
-            data: Union[np.integer, np.floating, np.bool_],
-            ) -> Union[int, float, bool]:
-        if isinstance(data, np.integer):
-            return int(data)
-        if isinstance(data, np.floating):
-            return float(data)
-        return bool(data)
-
-    def can_deserialize(self, data: Any) -> bool:
-        return False
-
-    def deserialize(self, data: Any) -> None:
-        raise NotImplementedError(
-            'Deserializing serialized Numpy scalars is not supported.')
-
-
 class UnitsSerializer(Serializer):
     """Serializer for data with units."""
 
     def __init__(self) -> None:
-        super().__init__(
-            name='units',
-            exclusive_types=(type(units.fg), type(1 * units.fg)))
+        super().__init__(name='units')
 
-    def can_serialize(self, data: Any) -> bool:
-        return isinstance(data, (Quantity, Unit))
+    class Codec(TypeEncoder):
```

Comment:
Can we call this `UnitCodec` or something similar to better communicate how it and `Codec2` (rename to `QuantityCodec`) are different?

Reply:
Addressed in 5394975.